### ISSUE TITLE
Add a `/count` endpoint for each `/search` endpoint

### ISSUE THREAD

**Issue by jonathanlukas** [camunda/camunda#19440]
## Description

In addition to the `/search` endpoint for each entity, a `/count` endpoint should be present that would return the count of results for the current query.

## Proposal

Every `/search` endpoint gets an equivalent `/count` endpoint that receives the same query body, but just returns a json object with a single field `count`:

```json
{"count": 1337}
```

## Decisions

_Will be filled in by the project DRIs after a peer review phase._

**Comment by tmetzke** [comment]
Like the idea, @jonathanlukas 👍

**Comment by tmetzke** [comment]
@aleksander-dytko, please determine if we want to define this from a general product perspective.

**Comment by tmetzke** [comment]
One thing we still have to define here, if we want this, is how to design the endpoints.
The most obvious option I currently see is to have dedicated `count` endpoints like `POST /v2/process-instances/count` to "create a count request" for a resource. Maybe there are other viable options to discuss, like:
* Add an attribute to search endpoints, e.g. `returnItems` in the `page` object which is true by default but can be turned off, so only the `totalItems` (aka count) is returned in the `page` object, for example.

**Comment by marcosgvieira** [comment]
For this one I believe we added the count on the result of a search as part of the page object response, would we be ok to close this one?

**Comment by tmetzke** [comment]
This information can indeed be deduced from the search response already.
However, this goes a bit further, as it's supposed to respond only with the count, without the list of items.

**Comment by tmetzke** [comment]
Plus, this could yield the **exact** count in ES/OS, which the search queries don't do for performance reasons. They only return 10,000 as the total value (ES/OS provide a hint that there is more which we want to incorporate into our API as well with https://github.com/camunda/camunda/issues/33885).

**Comment by jonathanlukas** [comment]
How about we make this endpoint a differentiator by using the real-time value (from zeebe) here? Query via database, count in real-time.

**Comment by tmetzke** [comment]
The runtime state doesn't have all the data that resides in secondary storage AND doesn't support the query options we have in the API. You would count and search for completely different data. If I wanted to count for all completed user tasks, the count on real-time data would return 0 because it doesn't contain that data.

**Comment by jonathanlukas** [comment]
@tmetzke totally right... I forgot about the history here...

**Comment by eppdot** [comment]
isn't this pretty much related (duplicate/covered-by) of [this](https://github.com/camunda/camunda/issues/34341) one?

**Comment by jonathanlukas** [comment]
I would say so. Can one fetch a page size of 0?

**Comment by tmetzke** [comment]
Yeah, I guess if the search endpoints are flexible enough, we don't need additional count endpoints. 

We could check the behavior of "size: 0" search requests on all data layers. I actually can't say what happens.

**Comment by jonathanlukas** [comment]
That would be great! We can then include this in our migration guide.

**Comment by tmetzke** [comment]
@jonathanlukas, is this something you have some time to test? I would love to look into this but I definitely won't have time before the code freeze next Tuesday 😄

**Comment by jonathanlukas** [comment]
@tmetzke unfortunately not... would that be a unit test to be added?

**Comment by tmetzke** [comment]
An integration/acceptance test that runs on all supported databases makes more sense I guess. Let's see: https://github.com/camunda/camunda/pull/36984

**Comment by tmetzke** [comment]
Looks like searching with a page limit of `0` yields the correct count via `page.totalItems` and no `items` (to save some bandwidth). It only yields correct results until the hit accuracy is reached, so https://github.com/camunda/camunda/issues/34341 would still be needed to get exact counts.
